src/main/java/com/iexec/core/chain/DealWatcherService.java

```diff
@@ -175,7 +190,36 @@ void replayDealEvent() {
     private Disposable subscribeToDealEventInRange(BigInteger from, BigInteger to) {
         log.info("Replay Watcher DealEvent started [from:{}, to:{}]",
                 from, (to == null) ? "latest" : to);
-        return iexecHubService.getDealEventObservable(from, to)
+        EthFilter filter = createDealEventFilter(from, to);
+        return iexecHubService.getDealEventObservable(filter)
+                .map(this::schedulerNoticeToDealEvent)
                 .subscribe(dealEvent -> dealEvent.ifPresent(this::onDealEvent));
     }
+
+    EthFilter createDealEventFilter(BigInteger from, BigInteger to) {
+        DefaultBlockParameter fromBlock = DefaultBlockParameter.valueOf(from);
+        DefaultBlockParameter toBlock = DefaultBlockParameterName.LATEST;
+        if (to != null) {
+            toBlock = DefaultBlockParameter.valueOf(to);
+        }
+        EthFilter filter = new EthFilter(fromBlock, toBlock, chainConfig.getHubAddress());
+        BigInteger poolAddressBigInt = Numeric.toBigInt(chainConfig.getPoolAddress());
+        filter.addSingleTopic(EventEncoder.encode(SCHEDULERNOTICE_EVENT));
+        filter.addSingleTopic(Numeric.toHexStringWithPrefixZeroPadded(poolAddressBigInt, 64));
+        return filter;
+    }
+
+    Optional<DealEvent> schedulerNoticeToDealEvent(IexecHubContract.SchedulerNoticeEventResponse schedulerNotice) {
+        dealEventsCount++;
+        BigInteger noticeBlockNumber = schedulerNotice.log.getBlockNumber();
+        if (latestBlockNumberWithDeal.compareTo(noticeBlockNumber) < 0) {
+            latestBlockNumberWithDeal = noticeBlockNumber;
+        }
+        log.info("Received new deal [dealEventsCount:{}, chainDealId:{}]", dealEventsCount, BytesUtils.bytesToString(schedulerNotice.dealid));
+        if (schedulerNotice.workerpool.equalsIgnoreCase(chainConfig.getPoolAddress())) {
+            return Optional.of(new DealEvent(schedulerNotice));
+        }
+        log.warn("This deal event should not have been received");
```

Comment:
Could we log the block number as well? With multi-threading, this log line could be way away from the `Received new deal` line.

Reply:
Sure, please see 2486787